src/ansys/aedt/core/modeler/circuits/primitives_maxwell_circuit.py

```diff
@@ -284,3 +284,87 @@ def create_winding(self, name=None, location=None, angle=0, use_instance_id_netl
         )
         id.set_property("Name", name)
         return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_isin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
+        """Create a sinusoidal current source.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the current source. The default is ``None``.
+        value : float, optional
+            Value for the amplitude of current. The default is ``1``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle of rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list or not. The default is ``False``.
+
+        Returns
+        -------
+        :class:`ansys.aedt.core.modeler.cad.object_3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateComponent
+        """
+        if location is None:
+            location = []
+        id = self.create_component(
+            name,
+            component_library="Sources",
+            component_name="ISin",
+            location=location,
+            angle=angle,
+            use_instance_id_netlist=use_instance_id_netlist,
+        )
+
+        id.set_property("Ia", value)
+        id.set_property("Name", name)
+        return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_vsin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
```

Comment:
all the comments in create_isin should be applied here too.

Reply:
I'll mark this as resolved since I'm applying the same changes to 'create_v_sin' as to 'create_i_sin'